### PR TITLE
ARROW-3751: [Gandiva][Python] Add more cython bindings for gandiva

### PR DESCRIPTION
**Pull request by suquark**
https://issues.apache.org/jira/projects/ARROW/issues/ARROW-3751

### CODE REVIEW

**Comment by pcmoritz** [comment]
This looks good to me! Can you also add a test?

File: python/pyarrow/gandiva.pyx [hunk 1]
```diff
@@ -241,6 +248,76 @@ cdef class TreeExprBuilder:
...
+        cdef c_string _v
+        for v in values:
+            _v = v.encode('UTF-8')
+            c_values.insert(_v)
+        r = TreeExprBuilder_MakeInExpressionString(node.node, c_values)
+        return Node.create(r)
+
+    def make_in_expression(self, Node node, values, dtype):
+        cdef DataType type = _as_type(dtype)
+        if type.id == _Type_INT32 or type.id == _Type_TIME32:
```
Comment (pcmoritz) [inline python/pyarrow/gandiva.pyx]: Don't the time and date cases need to be handled differently (calling TreeExprBuilder_MakeInExpressionDate etc.?)

File: python/pyarrow/tests/test_gandiva.py [hunk 2]
```diff
@@ -100,6 +100,60 @@ def test_filter():
...
+    filter = gandiva.make_filter(table.schema, condition)
+    result = filter.evaluate(table.to_batches()[0], pa.default_memory_pool())
+    assert list(result.to_array()) == [1, 2]
+
+    # string
+    builder = gandiva.TreeExprBuilder()
+    node_a = builder.make_field(table.schema.field_by_name("a"))
+    cond = builder.make_in_expression(node_a, ['an', 'nd'], pa.string())
+    condition = builder.make_condition(cond)
+
```
Comment (pcmoritz) [inline python/pyarrow/tests/test_gandiva.py]: Let's add tests for date and time stuff here too :)

**Comment by suquark** [comment]
Ready to merge. Just wait the CI.

**Review by pcmoritz — APPROVED** [review]

**Comment by pcmoritz** [comment]
Awesome, thanks for this! Do you want to create a JIRA with the issue and also link this PR so it is clear where it comes up?

**Comment by suquark** [comment]
Yes. I have mentioned `make_in_expression` in a new issue: https://issues.apache.org/jira/browse/ARROW-3799.

**Pull request closed without merging**